### PR TITLE
Show pending WCA ID claims to Seniors

### LINKED ISSUE #728: Allow Senior Delegates to see all outstanding WCA ID claims for their team
Tim McMahon requested "Would it be possible for Senior Delegates to see WCA ID requests "pending" with Delegates in their Region? I've had a couple of competitors complain. I'm not sure how many outstanding requests there are."

### PR DESCRIPTION
**Pull request by pedrosino**
Fixes #728 

Here's a sample:
image

### CODE REVIEW

**Comment by pedrosino** [comment]
@TimMcMahon is this what you had in mind?
@cubewhiz

**Comment by pedrosino** [comment]
Here's what the /users page would look like if the user has a pending WCA ID claim:
image

File: WcaOnRails/app/helpers/notifications_helper.rb [hunk 1]
```diff
@@ -31,6 +31,17 @@ def notifications_for_user(user)
...
     end
 
+    if user.senior_delegate?
+      user.subordinate_delegates.each do |delegate|
+        delegate.users_claiming_wca_id.where.not(confirmed_at: nil).each do |user_claiming_wca_id|
+          notifications << {
+            text: "#{user_claiming_wca_id.email} has claimed WCA ID #{user_claiming_wca_id.unconfirmed_wca_id} and is waiting for #{delegate.name}",
+            url: edit_user_path(user_claiming_wca_id.id, anchor: "wca_id"),
+          }
+        end
```
[the remark this answers is not in the document]
Comment (jonatanklosko) [reply WcaOnRails/app/helpers/notifications_helper.rb]: Alright, I'm fine with it then =)

File: WcaOnRails/app/helpers/notifications_helper.rb [hunk 2]
```diff
@@ -31,6 +31,17 @@ def notifications_for_user(user)
       end
     end
 
+    if user.senior_delegate?
+      user.subordinate_delegates.each do |delegate|
+        delegate.users_claiming_wca_id.where.not(confirmed_at: nil).each do |user_claiming_wca_id|
+          notifications << {
+            text: "#{user_claiming_wca_id.email} has claimed WCA ID #{user_claiming_wca_id.unconfirmed_wca_id} and is waiting for #{delegate.name}",
```
[the remark this answers is not in the document]
Comment (jonatanklosko) [reply WcaOnRails/app/helpers/notifications_helper.rb]: Not sure either. I think email does make sense, it happens that the same person claims a WCA ID from different accounts, if we use name then the given delegate would be confused by having two identical notifications.

File: WcaOnRails/app/assets/stylesheets/users.scss [hunk 3]
```diff
@@ -20,6 +20,13 @@ td.unconfirmed-email {
   font-style: italic;
 }
 
+.pending-id {
+  background-color: #f2dede;
+  border-color: #ebccd1;
+  color: #a94442;
```
Comment (jonatanklosko) [inline WcaOnRails/app/assets/stylesheets/users.scss]: These should go in the variables file (maybe you can even reuse existing colours).

File: WcaOnRails/app/assets/stylesheets/users.scss [hunk 4]
```diff
@@ -20,6 +20,13 @@ td.unconfirmed-email {
...
+  padding: 2px;
```
Comment (jonatanklosko) [inline WcaOnRails/app/assets/stylesheets/users.scss]: Any reason for that? As far as I can imagine, this makes claimed and unclaimed WCA IDs look differently in the table.

File: WcaOnRails/app/controllers/users_controller.rb [hunk 5]
```diff
@@ -36,7 +36,7 @@ def index
           total: users_array.size,
           rows: selected_rows.map do |user|
             {
-              wca_id: user.wca_id ? view_context.link_to(user.wca_id, person_path(user.wca_id)) : "",
+              wca_id: user.wca_id ? view_context.link_to(user.wca_id, person_path(user.wca_id)) : (user.unconfirmed_wca_id ? view_context.link_to(user.unconfirmed_wca_id, edit_user_path(user.id, anchor: "wca_id"), class: "pending-id") : ""),
```
Comment (jfly) [inline WcaOnRails/app/controllers/users_controller.rb]: For the record: I am firmly opposed to nested ternaries. I think it's always better to pull this sort of construct out into a separate variable or function or something.

**Comment by cubewhiz** [comment]
I agree with Jeremy--I do not think that all subordinate Delegate claims should appear on the Senior's notifications. It will be too numerous and confusing.

File: WcaOnRails/app/helpers/notifications_helper.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (pedrosino) [reply WcaOnRails/app/helpers/notifications_helper.rb]: Maybe name and email?

File: WcaOnRails/app/assets/stylesheets/users.scss (hunk 3 above)
Comment (pedrosino) [reply WcaOnRails/app/assets/stylesheets/users.scss]: Actually this was derived from the alert-danger. But that style makes the text too big, IIRC.

File: WcaOnRails/app/controllers/users_controller.rb (hunk 5 above)
Comment (pedrosino) [reply WcaOnRails/app/controllers/users_controller.rb]: Indeed. I tried to make an <%if in there but it didn't work. A variable would be better.

**Comment by pedrosino** [comment]
I was doing what @TimMcMahon asked in # 728.
I agree that a separate page for this would be better.

As for # 2, also came from Tim at https://github.com/thewca/worldcubeassociation.org/issues/728#issuecomment-241303637

I believe the idea is that if the user shows up in the search, the delegate would have a visual clue about pending claim. We can add a tooltip to explain what it means.

**Comment by jfly** [comment]
> As for # 2, also came from Tim at #728 (comment)
>
> I believe the idea is that if the user shows up in the search, the delegate would have a visual clue about pending claim. We can add a tooltip to explain what it means.

Can we pull that part of this PR out into a separate PR? I think this warrants its own, dedicated discussion.

**Comment by pedrosino** [comment]
Sorry for the lack of response.
I'll try to do this tomorrow morning.

**Comment by ronaldmansveld** [comment]
Any progress on this? Or should this PR be closed in favour of new PRs?

**Comment by viroulep** [comment]
@pedrosino: do you have time to work on the separate page?

I could take over if needed!
(kind of motivated by the fact the seniors just got an spreadsheet for this, and I think it would be much better if it was scoped with only our subordinate Delegates, and if the Board wouldn't have to recreate the list periodically ^^)

**Comment by viroulep** [comment]
I gave a shot at a page design in #3500!

**Comment by pedrosino** [comment]
Thanks, @viroulep 
I didn't have much time to work on this, and was actually focusing on the other part first.

**Comment by jfly** [comment]
I think thanks to @viroulep's work in #3500 means we can close this issue. @pedrosino, feel free to reopen or create a new PR if there's any work you'd like to continue with!

**Pull request closed without merging**